### PR TITLE
#1546 - Make Translation of LinearMap an AffineMap

### LINKED ISSUE #1546: Make Translation of LinearMap an AffineMap

### PR DESCRIPTION
**Pull request by mforets**
Closes #1546.

### CODE REVIEW

File: src/Translation.jl [hunk 1]
```diff
@@ -182,6 +182,9 @@ Unicode alias constructor ⊕ (`oplus`) for the lazy translation operator.
 # translation from the left
 ⊕(v::AbstractVector, X::LazySet) = Translation(X, v)
 
+# the translation of a lazy linear map is a (lazy) afine map
```
Comment (schillic) [inline src/Translation.jl]:
```suggestion
# the translation of a lazy linear map is a (lazy) affine map
```

File: test/unit_Translation.jl [hunk 2]
```diff
@@ -66,4 +66,9 @@ for N in [Float64, Rational{Int}, Float32]
     tr = Ball2(zeros(N, 2), N(1)) ⊕ N[1, 0]
     @test ρ(N[1, 0], tr) == N(2)
     @test_throws MethodError constraints_list(tr)
+
+    # the translation of a lazy linear map returns an affine map
+    M = N[1 0; 0 2]; B = BallInf(zeros(N, 2), N(1)); v = N[1, 0]
+    tr = M * B ⊕ v
+    @test tr isa AffineMap && tr.M == M tr.X == B && tr.v == v
```
Comment (schillic) [inline test/unit_Translation.jl]:
```suggestion
    @test tr isa AffineMap && tr.M == M && tr.X == B && tr.v == v 
```

**Comment by mforets** [comment]
The problem with this PR is that the `constraints_list` of an affine map call the constraints list of the associated translation.. which is again transformed to an `AffineMap` and this triggers an infinite recursion.

**Comment by schillic** [comment]
True, I did not think about this. I see two options:
1. Outsource the code of `constraints_list(::Translation)` to a helper function `constraints_list_translation(::LazySet, ::AbstractVector)` and call that function from `constraints_list(::AffineMap)`.
2. Add an optional argument to the constructor in this PR to circumvent the automatic conversion (and use it for `constraints_list(::AffineMap)`).

**Comment by schillic** [comment]
I implemented option 1.

**Comment by mforets** [comment]
Looks good! 👍

**Pull request merged**